TypeScript error with webpack 5.x

Getting the following error after updating from 4.6.1 to 4.7.0
```
Error: webpack.config.find-duplicates.ts(8,9): error TS2322: Type 'DuplicatesPlugin' is not assignable to type 'WebpackPluginInstance | ((this: Compiler, compiler: Compiler) => void)'.
  Type 'DuplicatesPlugin' is not assignable to type 'WebpackPluginInstance'.
    Types of property 'apply' are incompatible.
      Type '(compiler: ICompiler) => void' is not assignable to type '(compiler: Compiler) => void'.
        Types of parameters 'compiler' and 'compiler' are incompatible.
          Property 'plugin' is missing in type 'Compiler' but required in type 'ICompiler'
```

Here's a link to my PR that fails:
https://github.com/yoavain/webpack-playground/pull/401/checks?check_run_id=2185910775

--- a/HISTORY.md
+++ b/HISTORY.md
@@ -1,6 +1,11 @@
 History
 =======
 
+## UNRELEASED
+
+* Bug: Fix plugin types.
+  [#161](https://github.com/FormidableLabs/inspectpack/issues/161)
+
 ## 4.7.0
 
 * Feature: Include TypeScript definitions. (*[jensbodal][]*)

--- a/src/plugin/duplicates.ts
+++ b/src/plugin/duplicates.ts
@@ -13,9 +13,11 @@ import { numF, sort } from "../lib/util/strings";
 
 // Simple interfaces for webpack work.
 // See, e.g. https://github.com/TypeStrong/ts-loader/blob/master/src/interfaces.ts
+
+// Permissive compiler type that spans webpack v1 - current.
 interface ICompiler {
-  hooks: any;
-  plugin: (name: string, callback: () => void) => void;
+  hooks?: any;
+  plugin?: (name: string, callback: () => void) => void;
 }
 
 export interface ICompilation {
@@ -218,9 +220,11 @@ export class DuplicatesPlugin {
     if (compiler.hooks) {
       // Webpack4+ integration
       compiler.hooks.emit.tapPromise("inspectpack-duplicates-plugin", this.analyze.bind(this));
-    } else {
+    } else if (compiler.plugin) {
       // Webpack1-3 integration
       compiler.plugin("emit", this.analyze.bind(this) as any);
+    } else {
+      throw new Error("Unrecognized compiler format");
     }
   }
 

